src/test/kotlin/org/rust/ide/inspections/RsLiftInspectionTest.kt

```diff
@@ -0,0 +1,168 @@
+        }
+    """, checkWeakWarn = true, testmark = RsLiftInspection.Testmarks.insideRetExpr)
+
+    fun `test lift return in match 1`() = checkFixByText("Lift return out of 'match'", """
+        fn foo(x: bool) -> i32 {
+            /*weak_warning descr="Return can be lifted out of 'match'"*/match/*caret*//*weak_warning**/ x {
+                true => return 1,
+                false => return 0
+            }
+        }
+    """, """
+        fn foo(x: bool) -> i32 {
+            return match x {
+                true => 1,
+                false => 0
+            }
+        }
+    """, checkWeakWarn = true)
+
+    fun `test lift return in match 2`() = checkFixByText("Lift return out of 'match'", """
+        fn foo(x: bool) -> i32 {
+            /*weak_warning descr="Return can be lifted out of 'match'"*/match/*caret*//*weak_warning**/ x {
+                true => {
+                    return 1
+                },
+                false => {
+                    return 0;
+                }
+            }
+        }
+    """, """
+        fn foo(x: bool) -> i32 {
+            return match x {
+                true => {
+                    1
+                },
+                false => {
+                    0
+                }
+            }
+        }
+    """, checkWeakWarn = true)
+
+    fun `test lift return in match unavailable`() = checkFixIsUnavailable("Lift return out of 'if'", """
+        fn foo(x: bool) -> i32 {
+            match x/*caret*/ {
+                true => return 1,
+                false => {
+                    println("Oops");
+                }
+            }
+            return 0;
+        }
+    """, checkWeakWarn = true)
+
+    fun `test lift return from nested blocks`() = checkFixByText("Lift return out of 'if'", """
+        fn foo(x: bool, y: bool, z: bool) -> i32 {
+            /*weak_warning descr="Return can be lifted out of 'if'"*/if/*caret*//*weak_warning**/ x {
+                /*weak_warning descr="Return can be lifted out of 'match'"*/match/*weak_warning**/ y {
+                    true => return 1,
+                    false => {
+                        return 2;
+                    }
+                }
+            } else {
+                /*weak_warning descr="Return can be lifted out of 'if'"*/if/*weak_warning**/ z {
+                    return 3;
+                } else {
+                    {
+                        return 4
+                    }
+                }
+            }
+        }
+    """, """
+        fn foo(x: bool, y: bool, z: bool) -> i32 {
+            return if x {
+                match y {
+                    true => 1,
+                    false => {
+                        2
+                    }
+                }
+            } else {
+                if z {
+                    3
+                } else {
+                    {
+                        4
+                    }
+                }
+            }
+        }
+    """, checkWeakWarn = true)
+}
```

Comment:
Maybe add such tests as well: 
* inspection is unavailable when there is a `return` without expression
* comments after `return` doesn't affect inspection availability